### DISCUSSION TITLE
Prisma reverses manual changes made in migration file

### DISCUSSION THREAD

**Discussion by warrenbell** [prisma/prisma#19104] [Q&A]
I am trying to manually configure full text search for Postgresql. I have followed the instructions on how to manually modify migration files located at:

https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations

When I run prisma migrate with the manually modified migration file, Prisma comes back with another migration that reverses my changes.

Here is the manually modified file:

![Screenshot 2023-05-05 at 6 36 39 PM](https://user-images.githubusercontent.com/1256319/236588810-78153f3e-937e-4be3-8a4c-9518083d7205.png)

And here is the migration that Prisma creates to reverse it:

![Screenshot 2023-05-05 at 6 38 50 PM](https://user-images.githubusercontent.com/1256319/236588878-1bb55ae7-e006-48d8-8763-a00522078853.png)

I have done this same process to create triggers and Prisma does NOT reverse those changes. How do I get Prisma to let me add these full text features.

Thanks

**Comment by janpio** [comment]
1. I think your manual SQL is missing the `"` around `textSearch` to make sure the column is actually created uppercase. So Prisma currently tried to fix that for you, by dropping and recreating it. Adding the double quotes will probably get rid of the problem under `AlterTable`.
2. The dropping and re-creating of the index might already be solved by that as well, as that might just be the same underlying cause. If not,  let us know and we can take another look if you have to generate the GIN index differently.

**Reply by warrenbell** [reply]
I replaced textSearch with text_search and tried quotes around it. I got the same results. Prisma reversed the changes.

**Reply by janpio** [reply]
Can you share the SQL snippets as text so we can try to recreate this maybe? THanks.

**Comment by warrenbell** [comment]
I also tried to do introspecting by manually running the alter table directly in Postgres and then running prisma db pull. Prisma modifies the schema to this:

![Screenshot 2023-05-05 at 9 50 52 PM](https://user-images.githubusercontent.com/1256319/236598250-b84639d3-d155-4989-a096-79bc0fa2f2f4.png)

I then started with a blank db and ran migrate against it and I got a Postgres error:

![Screenshot 2023-05-05 at 9 52 41 PM](https://user-images.githubusercontent.com/1256319/236598318-3406d921-f3e1-4489-9718-34ef4e83d521.png)

**Reply by janpio** [reply]
Prisma does not support generated columns, and it picking them up as a default value is an unfortunate bug.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
